apps/vor/pkg/student/student.go

```diff
@@ -54,7 +55,65 @@ func authorizationMiddleware(s rest.Server, store Store) func(next http.Handler)
 		})
 	}
 }
+func registerClass(s rest.Server, store Store) http.Handler {
+	type responseBody struct {
+		Id          string     `json:"id"`
+		Name        string     `json:"name"`
+		DateOfBirth *time.Time `json:"dateOfBirth,omitempty"`
+	}
+	return s.NewHandler(func(w http.ResponseWriter, r *http.Request) *rest.Error {
+		id := chi.URLParam(r, "studentId")
 
+		student, err := store.Get(id)
+		if err != nil {
+			return &rest.Error{http.StatusNotFound, "Can't find student with specified id", err}
+		}
+
+		response := responseBody{
+			Id:          student.Id,
+			Name:        student.Name,
+			DateOfBirth: student.DateOfBirth,
+		}
+		if err := rest.WriteJson(w, response); err != nil {
+			return rest.NewWriteJsonError(err)
+		}
+		return nil
+	})
+}
+func registerAttendance(s rest.Server, store Store) http.Handler {
+	type requestBody struct {
+		StudentId string `json:"studentId"`
+		ClassId   string `json:"classId"`
+	}
+	return s.NewHandler(func(w http.ResponseWriter, r *http.Request) *rest.Error {
+		var requestBody requestBody
+		if err := rest.ParseJson(r.Body, &requestBody); err != nil {
+			return rest.NewParseJsonError(err)
+		}
+		attendance, err := store.InsertAttendance(requestBody.StudentId, requestBody.ClassId)
+		if err != nil {
+			return &rest.Error{http.StatusNotFound, "Can't create attendance", err}
+		}
+		if err := rest.WriteJson(w, attendance); err != nil {
+			return rest.NewWriteJsonError(err)
+		}
+		return nil
+	})
+}
+func getAttendance(s rest.Server, store Store) http.Handler {
```

Comment:
I think this endpoint would need to return all of the student's attendance data, including when they were absent.

Reply:
Okay then first we need to know how many session have been completed by that class. One way to check it is from the attendance table. Take all attendance for that classId, and select only the unique date. Is that enough?

Reply:
I think  that might not be good enough, see my other comments on the school.go file.